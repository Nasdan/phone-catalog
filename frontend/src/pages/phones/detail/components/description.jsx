import React from 'react';
import PropTypes from 'prop-types';
import {
  container,
  title,
  getCircleByColor,
  price,
  description,
} from './description.styles';

export const Description = ({ phone }) => (
  <div className={container}>
    <h3 className={title}>{phone.title}</h3>
    {phone.color && <p className={getCircleByColor(phone.color)}>Color </p>}
    <p className={description}>{phone.description}</p>
    <p className={price}>{phone.price}</p>
  </div>
);

Description.propTypes = {
  phone: PropTypes.object.isRequired,
};
