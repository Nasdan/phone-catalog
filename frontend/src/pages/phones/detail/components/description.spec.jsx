import React from 'react';
import { shallow } from 'enzyme';
import { Description } from './description';

describe('pages/phones/detail/components/description specs', () => {
  it('should render as expected when feeding phone', () => {
    // Arrange
    const phone = {
      title: 'test title',
      color: 'test color',
      description: 'test description',
      price: 'test price',
    };

    // Act
    const component = shallow(<Description phone={phone} />);

    // Assert
    expect(component).toMatchSnapshot();
  });
});
