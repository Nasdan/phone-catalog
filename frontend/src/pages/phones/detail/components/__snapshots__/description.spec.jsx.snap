// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/phones/detail/components/description specs should render as expected when feeding phone 1`] = `
<div
  className="css-19n6oew"
>
  <h3
    className="css-14pd6bw"
  >
    test title
  </h3>
  <p
    className="css-dhvz4c"
  >
    Color 
  </p>
  <p
    className="css-1kxrhf3"
  >
    test description
  </p>
  <p
    className="css-130s0i6"
  >
    test price
  </p>
</div>
`;
